config/src/main/java/tech/pegasys/pantheon/config/JsonUtil.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.config;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalLong;
+import java.util.function.Function;
+
+import com.fasterxml.jackson.core.JsonParser.Feature;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+public class JsonUtil {
+  public static OptionalLong getOptionalLong(final ObjectNode json, final String key) {
+    return getValue(json, key, JsonNode::asLong).map(OptionalLong::of).orElse(OptionalLong.empty());
+  }
+
+  public static <T> Optional<T> getValue(
```

Comment:
Have you considered Int, Long, String, and a boolean specific versions of getValue?  Having to pass the JsonNode::asFoo for each call feels a bit clunky.  We could then call the static method directly which should save a few cycles instead of passing an argument around and doing an indirect call.

Reply:
Yeah, that's probably worth doing.  I'll try reworking this method.

Reply:
Updated